nomad/client_agent_endpoint.go

```diff
@@ -149,24 +149,30 @@ func (a *Agent) monitor(conn io.ReadWriteCloser) {
 		return
 	}
 
-	currentServer := a.srv.serf.LocalMember().Name
-	var forwardServer bool
-	// Targeting a remote server which is not the leader and not this server
-	if args.ServerID != "" && args.ServerID != "leader" && args.ServerID != currentServer {
-		forwardServer = true
+	region := args.RequestRegion()
+	if region == "" {
+		handleStreamResultError(fmt.Errorf("missing target RPC"), helper.Int64ToPtr(400), encoder)
+		return
 	}
-
-	// Targeting leader and this server is not current leader
-	if args.ServerID == "leader" && !a.srv.IsLeader() {
-		forwardServer = true
+	if region != a.srv.config.Region {
+		// Mark that we are forwarding
+		args.SetForwarded()
 	}
 
-	if forwardServer {
-		a.forwardMonitorServer(conn, args, encoder, decoder)
-		return
+	// Handle serverID not equal to ours including region forwarding
```

Comment:
This comment is a little confusing since at this point we don't know if `serverID == this-server`. Perhaps:
```suggestion
	// Try to forward request to remote region/server
```